packages/tools/client-debug-view/src/components/data-object-views/SharedCounterView.tsx

```diff
@@ -37,12 +53,66 @@ export function SharedCounterView(props: SharedCounterViewProps): React.ReactEle
 		};
 	}, [sharedCounter, setValue]);
 
+
+    function decrementCounter(): void {
+        sharedCounter.increment(-1);
+	}
+
+    function incrementCounter(): void {
+		sharedCounter.increment(1);
+	}
+
+    const inputSetValue = (e: any): void => {
+        numberFormatCheck(e.target.value);
+    }
+
+    const numberFormatCheck = (val: number): void => {
+        const num = val ?? value;
+
+        if (!isFinite(num) || num[0] === '-') {
+            return
+        }
+
+        sharedCounter.increment(num - value);
+    }
+
 	return (
 		<Stack>
 			<StackItem>
 				<b>SharedCounter</b>
 			</StackItem>
-			<StackItem>Value: {value}</StackItem>
+			<StackItem>Value: {value} </StackItem>
+            <StackItem>
+                <TooltipHost
+					content="Decrement counter by 1 (min 0)."
+					id={decrementButtonTooltipId}
+				>
+                    <IconButton
+                            onClick={decrementCounter}
+                            disabled={value === 0}
+                            menuIconProps={{ iconName: "CalculatorSubtract" }}
+                            aria-describedby={decrementButtonTooltipId}
+                    />
+                </TooltipHost>
+
+                <TooltipHost
+                    content="Change the counter value by passing the number in input field"
+                    id={inputFieldTooltipId}
+                >
+                    <input type="number" onChange={inputSetValue} value={value}/>
```

Comment:
If we use the model I outlined in my other comment, we'll want the value here to default to `undefined`, so that users enter in their delta values in a blank entry form.

Reply:
We will also want to clear/reset the form when the user submits their increment/decrement.

Reply:
Removed the `value` filed.

Reply:
@Josmithr , Just to be clear. Do you want the input field to turn empty after the client clicks either one of the buttons?

Reply:
Yeah. I was sort of thinking that the default value we start the field in each time is `undefined`, and we can have placeholder text that says something like "Enter delta" or something like that to indicate the user should type a delta amount there. So, initialize the value to `undefined` and set it back to `undefined` each time we submit. Something like that.